actions/src/Scheduler.re

```diff
@@ -43,18 +46,82 @@ query MyQuery ($startTime: Int!, $endTime: Int!) {
 |}
 ];
 
+//// MUTATIONS /////
+module AddSyncItem = [%graphql
+  {|
+  mutation InsertBridgeSyncItem($bridgeId: Int!, $index: Int!, $startTime: Int!) {
+    insert_bridge_sync_one(object: {Info: "", bridge_id: $bridgeId, end_time: "", index: $index, start_time: $startTime, status: "Initiated"}) {
+      id
+    }
+  }
+|}
+];
+// Instead of an increment maybe we want this as a set.
+// Then if an arweave upload fails, we could set the sync time back to now to try again "immediately"
+module UpdateNextSync = [%graphql
+  {|
+mutation ScheduleNextSyncTime($id: Int!, $frequencyDurationSeconds: Int!) {
+  update_bridge_data_by_pk(pk_columns: {id: $id}, _inc: {next_scheduled_sync: $frequencyDurationSeconds}) {
+    id
+  }
+}
+|}
+];
+
+let addSyncItem = (bridgeId, index, startTime) => {
+  Client.instance
+  ->ApolloClient.mutate(
+      ~mutation=(module AddSyncItem),
+      {bridgeId, index, startTime},
+    )
+  ->ignore;
+  // TODO: id of bridgeSyncItem returned, so can be passed on to uploadChunkToArweave function
+};
+
+let updateSyncTime = (id, frequencyDurationSeconds) => {
+  Client.instance
+  ->ApolloClient.mutate(
+      ~mutation=(module UpdateNextSync),
+      {id, frequencyDurationSeconds},
+    )
+  ->ignore;
+};
+
+// for every item.
+// create a sync Item In the table.
+// mutation to update when next sync should be (based on frequency). (does this change if current sync fails.)
+// call uploadChunkToArweave
 let processBridges = () => {
   let currentTimestamp = Js.Math.floor_int(Js.Date.now() /. 1000.);
-  let startTime = currentTimestamp - 1800;
-  let endTime = currentTimestamp + 1800;
+  let endTime = currentTimestamp;
   Client.instance
   ->ApolloClient.query(
       ~query=(module GetNextScheduledSyncs),
-      {startTime, endTime},
+      {endTime: endTime},
+    )
+  ->Js.Promise.then_(
+      (result: ApolloClient__ApolloClient.ApolloQueryResult.t(_)) =>
+        GetNextScheduledSyncs.(
+          switch (result) {
+          | {data: Some({bridge_data})} =>
+            bridge_data
+            ->Array.map(({id, frequency_duration_seconds}) => {
+                addSyncItem(id, 1, currentTimestamp);
+                updateSyncTime(id, frequency_duration_seconds);
+                Upload.uploadChunkToArweave();
```

Comment:
Above this we need to add something like `FetchDataFromSkale` and have that return a promise so that this Upload waits for that command to finish.

And in-between it should update the status in the DB.